File: examples/typegraphs/kv.py
# skip:start
from typegraph import Graph, Policy, typegraph
from typegraph.graph.params import Cors
from typegraph.runtimes.kv import KvRuntime

# skip:end


@typegraph(
    # skip:start
    cors=Cors(allow_origin=["https://metatype.dev", "http://localhost:3000"]),
    # skip:end
)
def key_value(g: Graph):
    kv = KvRuntime("REDIS")

    g.expose(
        Policy.public(),
        get=kv.get(),
        set=kv.set(),
        delete=kv.delete(),
        keys=kv.keys(),
        values=kv.values(),
    )

File: examples/typegraphs/kv.ts
// skip:start
import { Policy, typegraph } from "@typegraph/sdk";
import { KvRuntime } from "@typegraph/sdk/runtimes/kv";

// skip:end

export const tg = await typegraph(
  {
    name: "key-value",
    // skip:next-line
    cors: { allowOrigin: ["https://metatype.dev", "http://localhost:3000"] },
  },
  (g) => {
    const kv = new KvRuntime("REDIS");
    const pub = Policy.public();
    g.expose({
      get: kv.get(),
      set: kv.set(),
      delete: kv.delete(),
      keys: kv.keys(),
      values: kv.values(),
    }, pub);
  },
);
